### PR TITLE
feat: keep capabilities in model while ensuring default capabilities

### PR DESCRIPTION
**Pull request by MikeTschudi**
1. Description:

Function `ensureRequiredSiteProperties` sets the capabilities in a model, deleting any capabilities already in the model. With this PR, the function ensures that the required properties are present without deleting capabilities that are not required.

1. Instructions for testing:

Run unit tests

1. Closes Issues: #<number> (if appropriate)

https://github.com/Esri/solution.js/issues/1050

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.
 
@rweber-esri @dbouwman

### CODE REVIEW

**Review by rweber-esri — APPROVED** [review]
This looks good to me. May want to wait for @dbouwman to take a peek too.

**Review by dbouwman — APPROVED** [review]
This is great! Thanks @MikeTschudi and @rweber-esri !

File: packages/sites/src/ensure-required-site-properties.ts [hunk 1]
```diff
@@ -68,7 +68,12 @@ export function ensureRequiredSiteProperties(
     "underlinedLinks",
     "globalNav",
   ];
-  if (!isPortal) {
+  const caps = (model.data.values.capabilities || []).reduce(
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/sites/src/ensure-required-site-properties.ts:71]: `data.values` really should be present... but putting in a `?` a la `const caps = (model.data.values?.capabilities || []).reduce(` is another option

[the remark this answers is not in the document]
Comment (MikeTschudi) [reply packages/sites/src/ensure-required-site-properties.ts:71]: If we're not sure, we need to put the check much higher in the function.  Lines 28 and 29 in the file are
```
  model.data.values.updatedAt = new Date().toISOString();
  model.data.values.updatedBy = username;
```
My unit tests for coverage failed here, long before I got to `model.data.values.capabilities`.

[the remark this answers is not in the document]
Comment (rweber-esri) [reply packages/sites/src/ensure-required-site-properties.ts:71]: Good point, I see `model.data.values` referenced above. I think that eliminates my concerns.

**Pull request merged**